### PR TITLE
chore: parse addPreloadScript params

### PR DESCRIPTION
**Pull request by Lightning00Blade**
WPT should cover this case https://github.com/web-platform-tests/wpt/pull/41932

Tracked at #1353

### CODE REVIEW

File: src/bidiMapper/domains/script/ScriptProcessor.ts [hunk 1]
```diff
@@ -50,6 +51,21 @@ export class ScriptProcessor {
   async addPreloadScript(
     params: Script.AddPreloadScriptParameters
   ): Promise<Script.AddPreloadScriptResult> {
+    if (params.contexts) {
+      if (params.contexts.length === 0) {
```
Comment (OrKoN) [inline src/bidiMapper/domains/script/ScriptProcessor.ts:55]: if zod validates this, we could remove this check

**Review by OrKoN — APPROVED** [review]

File: src/bidiMapper/domains/script/ScriptProcessor.ts (hunk 1 above)
Comment (Lightning00Blade) [reply src/bidiMapper/domains/script/ScriptProcessor.ts:55]: Sadly the auto-generated `zod` validator don't validate the one or more in `[+Object]` from CDDL, I will take a peak if its possible to implement. 

CC @jrandolf

Comment (thiagowfx) [reply src/bidiMapper/domains/script/ScriptProcessor.ts:55]: Can we file a FR here? https://github.com/google/cddlconv

I swear I had filed one already, but I cannot find it.

File: src/bidiMapper/domains/script/ScriptProcessor.ts [hunk 2]
```diff
@@ -50,6 +51,21 @@ export class ScriptProcessor {
...
+        throw new InvalidArgumentException('Invalid empty contexts list.');
```
Comment (thiagowfx) [inline src/bidiMapper/domains/script/ScriptProcessor.ts:56]: How about: "Contexts list is empty".

Invalid is implicit in the error type, we don't need to repeat it in the error message

File: src/bidiMapper/domains/script/ScriptProcessor.ts [hunk 3]
```diff
@@ -50,6 +51,21 @@ export class ScriptProcessor {
...
+      }
+
+      for (const contextId of params.contexts) {
+        const context = this.#browsingContextStorage.getContext(contextId);
+        if (!context.isTopLevelContext()) {
+          throw new InvalidArgumentException(
+            `addPreloadScript only supported on the top-level contexts, ${contextId} given.`
```
Comment (thiagowfx) [inline src/bidiMapper/domains/script/ScriptProcessor.ts:63]:
```suggestion
            `addPreloadScript is only supported on top-level contexts, but was '${contextId}' given.`
```

**Review by thiagowfx — COMMENTED** [review]
Add at least one test? It could be E2E. Unless it's already covered in WPT.

File: src/bidiMapper/domains/script/ScriptProcessor.ts (hunk 1 above)
Comment (Lightning00Blade) [reply src/bidiMapper/domains/script/ScriptProcessor.ts:55]: Filed - https://github.com/google/cddlconv/issues/16

Comment (thiagowfx) [reply src/bidiMapper/domains/script/ScriptProcessor.ts:55]: Thanks

**Comment by Lightning00Blade** [comment]
WPT test will cover all the use cases for invalid, I would prefer not to repeat them.

File: src/bidiMapper/domains/script/ScriptProcessor.ts [hunk 4]
```diff
@@ -50,6 +51,22 @@ export class ScriptProcessor {
...
+      if (params.contexts.length === 0) {
+        throw new InvalidArgumentException('Contexts list is empty.');
+      }
+
+      for (const contextId of params.contexts) {
+        const context = this.#browsingContextStorage.getContext(contextId);
+        if (!context.isTopLevelContext()) {
+          throw new InvalidArgumentException(
+            `Non top-level contexts given '${contextId}'.`
+          );
```
Comment (thiagowfx) [inline src/bidiMapper/domains/script/ScriptProcessor.ts:65]:
```suggestion
            `Non top-level context '${contextId}' given.`
          );
```

**Review by thiagowfx — APPROVED** [review]

File: src/bidiMapper/domains/script/ScriptProcessor.ts [hunk 5]
```diff
@@ -50,6 +51,22 @@ export class ScriptProcessor {
   async addPreloadScript(
     params: Script.AddPreloadScriptParameters
   ): Promise<Script.AddPreloadScriptResult> {
+    if (params.contexts) {
+      // Remove once https://github.com/google/cddlconv/issues/16 is implemented
```
Comment (thiagowfx) [inline src/bidiMapper/domains/script/ScriptProcessor.ts:55]:
```suggestion
      // XXX: Remove once https://github.com/google/cddlconv/issues/16 is implemented
```

**Comment by thiagowfx** [comment]
Added Bug: #1353

**Pull request merged**